File: movieApp/app/src/main/java/com/lbc/practice/movieapp/model/resource/MovieDataSource.java
package com.lbc.practice.movieapp.model.resource;

import android.content.Context;

import com.lbc.practice.movieapp.model.PlayingMovieResult;

import java.util.List;

public interface MovieDataSource {
    interface LoadDataCallBack {
        void onLoadData(List<PlayingMovieResult.ResultsBean> movieInfo);
        void onFailData(String errorMsg);
    }

    void seeMovieInfo(String apiKey,String language, int page, LoadDataCallBack loadDataCallBack);

    void favoriteMovieInfo(Context context, LoadDataCallBack loadDataCallBack);

}

File: movieApp/app/src/main/java/com/lbc/practice/movieapp/data/resource/MovieDataSource.kt
package com.lbc.practice.movieapp.data.resource;

import android.content.Context;

import com.lbc.practice.movieapp.data.PlayingMovieResult;


interface MovieDataSource {
    interface LoadDataCallBack {
        fun onLoadData(movieInfo: kotlin.collections.List<PlayingMovieResult.ResultsBean>)
        fun onFailData(errorMsg: String)
    }

    fun seeMovieInfo(apiKey: String, language: String, page: Int, loadDataCallBack: LoadDataCallBack)

    fun favoriteMovieInfo(context: Context, loadDataCallBack: LoadDataCallBack)

}
